Issue #27 - Search new stock name, ignore column and email support.

reynoldtan opened the pull request:
NOTE
          1. this update requires an update to KP nodes to include all
hook_alter() for this update to apply (Issue #6 - Rawphenotypes hook_alter() implementations).
          2. Should update code when - Issue #12 Stock names look-up should be restricted to an organism is merged.

- Upload functionality can now search new name (using a token eg. AGL)
to a specified project name.
- Ignore or skip a set column names.
- Provide a support email for data collectors to email for support.

To test:
A. Support email:
  In the homepage of you dev, in rawphenotypes dashboard, click need help? link and test 
  1. help topic select jump menu and 
  2. support email (please click if it starts a blank email or right click on the link to copy the email address to manually compose an email)

B. Origin
  1. In standard spreadsheet file, add a column Origin.
  2. Upload the spreadsheet file. Ensure to upload file into AGILE project.
     In all upload stages, it should skip this column.
  3. When done, download dataset to verify.

C. New stock names (with AGL).
  1. Upload to AGILE project a spreadsheet file where the name column contains stock names without AGL token.
  2. When complete, download dataset to verify the use of new stock names.

carolyncaron commented:
### A. Support Email
- I do see the help topic dropdown, and all the options appear to work as expected. However, I can't seem to find a link for support email - I assume I have to set this up through administration but can't find an option for it.
- Also, should we consider making the email appear only on the upload page? I feel like having it in the Raw pheno panel on the homepage will either get missed entirely by those who need it or spammed by those who haven't even tried...

### B. Origin
- I verified that an 'Origin' column is ignored when uploaded to AGILE.  Plus, 'Origin' appears as a new trait for AGILE: Preliminary Trials :+1:
- One thing I noticed is that when running the job manually, the terminal is _extremely_ verbose. Every stock ID and stock name is printed, in addition to progress and whenever a column is skipped (one print statement for each row). If this is for debugging purposes, we probably don't need all of the print statements anymore!
- Another thing is when uploading a file, the initial check for stock IDs appears to take a long time, thus the progress bar during Step 3 remains at "Waiting..." for half the duration of the job.

### C. New Stock Names (with AGL)
- I verified that when uploading to AGILE, stock names gained the AGL suffix when it was missing (gained nothing if it was already there).
- For a non-AGILE project, AGL was accepted when present but not added when missing.
- This functionality looks ready to me :)

Regarding the merge conflicts, we will need @laceysanderson's help with addressing this, as we suspect it has to do with recent updates to the module for Tripal 3 support.

reynoldtan commented:
- Removed print command that prints stock_id to terminal
- Added link to support email and video demo in Upload Page
![support_email](https://user-images.githubusercontent.com/15472253/37225311-d6baf17e-239b-11e8-8fbf-8712247da2a2.jpg)

carolyncaron commented:
- I am still seeing excessive print statements when I run a job. My test case is the entire LDP panel (3 reps) with an origin column (value = Mars). Here is a snippet of what I see (the actual output is hundreds of lines):
![image](https://user-images.githubusercontent.com/7927760/37373473-4f4e6dda-26dc-11e8-9fc8-9dbcee177979.png)
- I couldn't see the support email on either the front page or upload form, but after some digging I noticed I was an additional commit ahead on the relevant branch for kp_nodes as compared to the last time I tested (see commit: [0a9ed89](https://github.com/UofS-Pulse-Binfo/kp_nodes/commit/0a9ed89f5a7fdff2531aaf62014cf00503185187)) It would have been helpful to know that I needed to set the email address! Lacey suggests perhaps setting a default email to be the same as admin_0? We then need to update the documentation to explain how to set the email for support (I think this needs a new issue - this reminds me that the docs also need to be updated regarding the update to handle environmental data)

laceysanderson commented on rawpheno.module:
```diff
@@ -326,6 +326,10 @@ function rawpheno_preprocess_rawpheno_upload_form_master(&$variables, $hook) {
   $variables['page_title'] = variable_get('rawpheno_upload_title');
+  // Support Email.
+  $support_email = '';
+  drupal_alter('rawpheno_support_email', $support_email);
```
There should be an API function for retrieving the support email. Furthermore, the code in kp_nodes determining the default support email (https://github.com/UofS-Pulse-Binfo/kp_nodes/pull/7/files#diff-ea83b3e1886aaaeac403bd6114856bd6R553)  should be in this rawphenotypes API function --otherwise only we are given a default ;-)

laceysanderson commented:
Note: I added a pull request for Kp Nodes as well: https://github.com/UofS-Pulse-Binfo/kp_nodes/pull/7#pullrequestreview-104331711

There are suggested changes on it too :-)

laceysanderson commented:
- Confirmed minimal printing to the terminal when upload jobs are run 👍 
- support email: this functionality is proposed to be fixed in #36 so I will test it there.
- origin & new stock names were already confirmed working by @carolyncaron 

Thus, I'm calling this PR Approved 🎉
